### PR TITLE
chore(nodebuilder/tests): Clean up sync_test tests

### PR DESCRIPTION
**Pull request by renaynay**
This PR is a part of the swamp cleanussy effort to clean up our integration tests and make them more deterministic + readable. 

This PR focuses on `sync_test.go` file.

It also contains a change to fraud proof broadcasting test but I can happily split out into separate PR.

### CODE REVIEW

**Review by distractedm1nd — APPROVED** [review]

File: nodebuilder/tests/swamp/swamp.go [hunk 1]
```diff
@@ -324,18 +324,26 @@ func (s *Swamp) remove(rn *nodebuilder.Node, sn []*nodebuilder.Node) ([]*nodebui
...
+	require.NoError(t, s.Network.DisconnectPeers(peerA.Host.ID(), peerB.Host.ID()))
+}
+
+func WithTrustedPeers(t *testing.T, cfg *nodebuilder.Config, trustedPeers ...*nodebuilder.Node) {
+	for _, trusted := range trustedPeers {
+		addrs, err := peer.AddrInfoToP2pAddrs(host.InfoFromHost(trusted.Host))
+		require.NoError(t, err)
+		cfg.Header.TrustedPeers = append(cfg.Header.TrustedPeers, addrs[0].String())
+	}
 }
```
Comment (Wondertan) [inline nodebuilder/tests/swamp/swamp.go:349]: Let's set trusted or bootstrap peers for nodes automatically. We should still allow users to set trusted peers if their test case requires it, but that should not be necessary to startup a healthy node.

Comment (renaynay) [reply nodebuilder/tests/swamp/swamp.go]: https://github.com/celestiaorg/celestia-node/pull/2162/commits/2883a67bb299e5c5d4aea5e9aef4cd131d6d6698

File: nodebuilder/tests/sync_test.go [hunk 2]
```diff
@@ -17,61 +16,184 @@ import (
...
-	h, err := bridge.HeaderServ.WaitForHeight(ctx, 20)
+	h, err := bridge.HeaderServ.WaitForHeight(ctx, numBlocks)
 	require.NoError(t, err)
+	require.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, numBlocks))
+
+	t.Run("light sync against bridge", func(t *testing.T) {
+		// create a light node that is connected to the bridge node as
+		// a bootstrapper
+		cfg := nodebuilder.DefaultConfig(node.Light)
+		light := sw.NewNodeWithConfig(node.Light, cfg)
```
Comment (Wondertan) [inline nodebuilder/tests/sync_test.go:70]: Can use NewLight constructor.

**Review by distractedm1nd — APPROVED** [review]

File: nodebuilder/tests/sync_test.go [hunk 3]
```diff
@@ -265,54 +348,56 @@ Steps:
...
 	err := bridge.Start(ctx)
 	require.NoError(t, err)
-
-	h, err := bridge.HeaderServ.WaitForHeight(ctx, 20)
-	require.NoError(t, err)
-
-	assert.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, 20))
-
-	addrs, err := peer.AddrInfoToP2pAddrs(host.InfoFromHost(bridge.Host))
+	err = bridge.HeaderServ.SyncWait(ctx)
```
[the remark this answers is not in the document]
Comment (renaynay) [reply nodebuilder/tests/sync_test.go]: changed back for some tests https://github.com/celestiaorg/celestia-node/pull/2162/commits/51b16a337d373ebf9cb8f9f1bea7fac92188d4fa

File: nodebuilder/tests/sync_test.go (hunk 2 above)
Comment (renaynay) [reply nodebuilder/tests/sync_test.go]: done

**Review by Wondertan — APPROVED** [review]
LGTM
Thank you for doing this!

**Pull request merged**